nixos/modules/services/web-apps/convos.nix

```diff
@@ -0,0 +1,66 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.convos;
+in
+{
+  options.services.convos = {
+    enable = mkEnableOption "Convos";
+    listenPort = mkOption {
+      type = types.port;
+      default = 3000;
+      example = 8080;
+      description = "Port the web interface should listen on";
+    };
+    listenAddress = mkOption {
+      type = types.str;
+      default = "*";
+      example = "127.0.0.1";
+      description = "Address or host the web interface should listen on";
+    };
+    reverseProxy = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enables reverse proxy support. This will allow Convos to automatically
+        pick up the <literal>X-Forwarded-For</literal> and
+        <literal>X-Request-Base</literal> HTTP headers set in your reverse proxy
+        web server. Note that enabling this option without a reverse proxy in
+        front will be a security issue.
+      '';
+    };
+  };
+  config = mkIf cfg.enable {
+    systemd.services.convos = {
+      description = "Convos Service";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "networking.target" ];
+      environment = {
+        CONVOS_HOME = "%S/convos";
+        CONVOS_REVERSE_PROXY = if cfg.reverseProxy then "1" else "0";
+        MOJO_LISTEN = "http://${toString cfg.listenAddress}:${toString cfg.listenPort}";
+      };
+      serviceConfig = {
+        ExecStart = "${pkgs.convos}/bin/convos daemon";
+        Restart = "on-failure";
+        StateDirectory = "convos";
+        WorkingDirectory = "%S/convos";
+        DynamicUser = true;
+        MemoryDenyWriteExecute = true;
+        ProtectSystem = "strict";
+        ProtectHome = true;
```

Comment:
@stigtsp at least a few of these options are already implied with `DynamicUser`. A quick search for `DynamicUser` in the `systemd` manual will cut a few lines here.

Reply:
Yes, `ProtectSystem` and `ProtectHome` are implied and can be removed.

Reply:
Thanks @talyz @aanderse :)  I've removed `ProtectSystem=strict`.

Seems like `ProtectHome=read-only` is implied by `DynamicUser=true` according to the documentation, so keeping `ProtectHome=true`.

Added `SystemCallFilter`, `SystemCallArchitectures`, `CapabilityBoundingSet`, and some more flags highlighted by `systemd-analyze security`.

```
→ Overall exposure level for convos.service: 1.3 OK 🙂
```

Does this look ok?

Reply:
Ah, yes, you're right :) Looks good to me, assuming all features you want still work as expected ;)